### DISCUSSION TITLE
Neutral placement of empty date fields

### DISCUSSION THREAD

**Discussion by burningT** [TechbeeAT/jtxBoard#1331] [Ideas]
At present, if e.g. tasks are ordered by date, tasks with empty date fields are placed at the end of the list. This places them visually in the very far future. That becomes odd if e.g. a task only has start date now while the tasks are ordered by start and secondarily by due date. It results in the task getting placed out of sight even if all due dates are far from now. Wouldn't it be more helpfull  to consider empty dates neutral and place them as if they contain the date now?

**Comment by patrickunterwegs** [comment]
Hi @burningT !
Thanks for bringing this up in a discussion. Basically there have been discussions on this topic what would be the expected sort order. You could argue in both directions. Some say, that when you have a task without a date, you don't care when it's done, it has no priority, so you don't want to have it on top of the list. 
Your case is the other one. You'd like to see them on the top because otherwise you might miss them as they are at the end...

I don't know what's better. Any other opinions?

**Reply by burningT** [reply]
Just to make sure: My idea is not exactly to place items without date simply on top of the list. I think it is sensible to have items e.g. with a due date in the past to be shown first. However, I also think that items with a due date in the future should be shown after those without entry. You could say I would place items without entry in the middle. Thats why I called it neutral.

**Reply by patrickunterwegs** [reply]
Well, that's a bit difficult, how would you break this down technically? 
Would you place the elements without date then after the ones due in a week? I'm a month? Any other? If those entries end up in between, I'm afraid that it would look rather random for end users...

**Reply by burningT** [reply]
Sure, random placement is even less usable. Thats why I suggest considering empty dates as if they would contain the time now. Thats seems both technically solvable and greatly usable to me. Isnt it?

**Comment by rogercreagh** [comment]
I can see three cases where you might not want to set a due date.
1. where you don't care when you start it, In effect it is in a separate queue of "awaiting start"
2. where you are starting it but it doesn't matter when you get it done by
3. where it is dependent on another task completing so you don't want to set a due (or start) date till you have completed the dependencies

In both cases 1 & 2 the effective due date is far in the future so they should be at the end of the list. Once you are working on the task you can set a due date. You can also use progress, categories and status (include jtxB extended statuses) to sort/filter the list

When I was using jtxB for tasks I used Kanban view and an extended status of "Dependent" to handle the third case. Once the dependencies were fulfilled I would set start and due date.

So for me undated at end of the due date list makes sense.

**Reply by burningT** [reply]
Sounds interesting. So you would always start and complete all tasks with due date before starting tasks without due date?

**Reply by rogercreagh** [reply]
No. Probably set the due date when you start the task (and move it to the appropriate column and update the status). Although there is no need to set the due date when starting, you might be working on it with no definite due date so it makes sense to have it after all the ones that need to be completed by a specific date. Or set a target date far enough in the future that it comes after the ones with firm dates but before the not started ones.

**Reply by burningT** [reply]
Oh I see. Infact I am trying to be clean with my due dates to prevent getting lost in true dates and work around dates. And apart from that, I think it would solve the use case I have in mind only partially.

I am looking for the ability to have a brief list in the widget and app, showing me the tasks being relevant now or becoming relevant next. The list should start with the most urgent task. I think the most urgent task could be figured out here by finding the earliest start and due date. However, the list only remains consistent, if an empty date is considered as if it contains the date now. Or am I missing another option?

**Reply by rogercreagh** [reply]
Why not use the priority setting - there are 10 levels of priority which ought to be granular enough to be able to filter the most relevant ones. Just walk down the priorities until you have enough for your list?

You can change the priority of a task at any time - I find sometimes it is better to use your own judgement as to what is most relevant rather than rely on an algorithm that doesn't know that client A is more tolerant of slipping dates than client B for example. Or that you have a cool idea for task X that means you want to get onto it before task Y which has an earlier due date. Or whatever. Life is judgement calls, and an algorithm is never going to do that.

**Reply by burningT** [reply]
You make an interesting point here. Quite possibly this is a more challenging topic than I initially thought. 

This becomes clear to me as your approach works fine for you but wouldn't for me. In my imagination the list of relevant tasks is a 1 D representation of the Eisenhower method. Or at least one dimension of it. Urgency results from start and due date and the importance is represented by the priority.

Hence, in my case the priority is already in use and the combination of both dates needs to represent the urgency in the way I described above.

Now, this seems like different use cases. I am wondering, is there a way to allow both?
